### PR TITLE
Block executing T-SQL trigger when there is any Postgres' function, procedure or trigger is in the call stack.

### PR DESCRIPTION
**Pull request by Deepesh125**
### Description
Recently, we discovered crash occurring due to double freeing of composite trigger data. Fix is done as part of https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/154 and this commit adds test cases for the same.

This commit also blocks executing T-SQL trigger when there is any Postgres' function, procedure or trigger is in call stack.

Task: BABEL-4220
Signed-off-by: Dipesh Dhameliya <dddhamel@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/TestTriggerInteroperability.out [hunk 1]
```diff
@@ -0,0 +1,170 @@
+-- tsql
```
Comment (surendravishnoi) [inline test/JDBC/expected/TestTriggerInteroperability.out:1]: Please move these tests to input/interoperability/TestProcedureWithTriggers.mix

File: test/JDBC/expected/TestTriggerInteroperability.out [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+create table test_tbl_trig_bbf_1(a int);
+create table test_tbl_trig_bbf_2(a int);
+GO
+
+-- psql
```
Comment (surendravishnoi) [inline test/JDBC/expected/TestTriggerInteroperability.out:6]: we want to add these test cases with and without error throw (ignore if they are already part of TestProcedureWithTriggers.mix)
begin TSQL insert -> TSQL trigger -> PG trigger (error)
begin TSQL insert -> PG trigger -> TSQL trigger -> PG trigger (error)
begin TSQL insert -> TSQL trigger -> PG trigger -> TSQL trigger (error)

File: contrib/babelfishpg_tsql/src/pl_handler.c [hunk 3]
```diff
@@ -32,6 +32,7 @@
 #include "commands/defrem.h"
 #include "commands/sequence.h"
 #include "commands/tablecmds.h"
+#include "commands/trigger.h"
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/pl_handler.c:35]: why this header?

File: test/JDBC/expected/TestTriggerInteroperability.out [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+select count(*) from psql_tbl_1;
+GO
+~~START~~
+int
+0
+~~END~~
+
+
+
+-- begin TSQL insert -> PG trigger -> TSQL trigger -> PG trigger (error)
```
Comment (surendravishnoi) [inline test/JDBC/expected/TestTriggerInteroperability.out:370]: let us add a test to check PG proc -> insert -> TSQL trigger
also, TSL proc -> insert -> PG trigger -> TSQL trigger

File: contrib/babelfishpg_tsql/src/pl_handler.c (hunk 3 above)
Comment (Deepesh125) [reply contrib/babelfishpg_tsql/src/pl_handler.c:35]: For the hook

File: test/JDBC/expected/TestTriggerInteroperability.out (hunk 4 above)
Comment (Deepesh125) [reply test/JDBC/expected/TestTriggerInteroperability.out:370]: Added

**Review by surendravishnoi — APPROVED** [review]

**Pull request merged**